alternator/conditions.cc

```diff
@@ -67,18 +68,28 @@ comparison_operator_type get_comparison_operator(const rjson::value& comparison_
     return it->second;
 }
 
+using cql3::operator_type;
+
 static ::shared_ptr<cql3::restrictions::single_column_restriction::contains> make_map_element_restriction(const column_definition& cdef, std::string_view key, const rjson::value& value) {
     bytes raw_key = utf8_type->from_string(sstring_view(key.data(), key.size()));
     auto key_value = ::make_shared<cql3::constants::value>(cql3::raw_value::make_value(std::move(raw_key)));
     bytes raw_value = serialize_item(value);
     auto entry_value = ::make_shared<cql3::constants::value>(cql3::raw_value::make_value(std::move(raw_value)));
-    return make_shared<cql3::restrictions::single_column_restriction::contains>(cdef, std::move(key_value), std::move(entry_value));
+    auto r = make_shared<cql3::restrictions::single_column_restriction::contains>(cdef, key_value, entry_value);
+    using namespace cql3::restrictions::wip;
+    r->expression = binary_operator{
+        std::vector{column_value(&cdef, std::move(key_value))}, &operator_type::EQ, std::move(entry_value)};
```

Comment:
operator_type::CONTAINS?

Reply:
No.  WIP uses CONTAINS only for its original purpose of expressing "collection contains a value x".  For the purpose above ("map[key]=value"), WIP uses EQ.  The old code uses CONTAINS for both purposes, which I think is unorthodox.

Reply:
You are correct. According to the comment, it uses "contains" because the structure has all the necessary members.